bftengine/tests/simpleStorage/MetadataStorageTypes.hpp

```diff
@@ -46,8 +47,8 @@ struct MetadataObjectInfo {
 
 // Object data to be stored for every write operation in transaction.
 struct RequestInfo {
```

Comment:
It'd probably be easier to just use a vector rather than unique_ptr + len here.

I'm also not sure if you should take this by rvalue reference `&&buf` rather than buf. It's a tricky c++ area. It may be easier, and clearer to just pass the unique_ptr by value also. That shows that ownership is being given to the RequestInfo struct. Passing by reference, and looking at the constructor signature doesn't imply that's what's happening.

Reply:
Changed it.